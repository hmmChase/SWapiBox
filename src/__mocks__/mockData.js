export const randomFilmCrawl = {
  crawl: 'Film Crawl',
  title: 'Movie Title',
  date: '1982-09-18'
};

export const cardObj1 = {
  name: 'T-16 skyhopper',
  model: 'T-16 skyhopper',
  class: 'repulsorcraft',
  passengers: '1'
};

export const cardObj2 = {
  name: 'Snowspeeder',
  model: 't-47 airspeeder',
  class: 'airspeeder',
  passengers: '0'
};

export const favorites = [cardObj1, cardObj2];

export const categoryData = [
  {
    name: 'Sand Crawler',
    model: 'Digger Crawler',
    class: 'wheeled',
    passengers: '30'
  },
  {
    name: 'T-16 skyhopper',
    model: 'T-16 skyhopper',
    class: 'repulsorcraft',
    passengers: '1'
  },
  {
    name: 'X-34 landspeeder',
    model: 'X-34 landspeeder',
    class: 'repulsorcraft',
    passengers: '1'
  },
  {
    name: 'TIE/LN starfighter',
    model: 'Twin Ion Engine/Ln Starfighter',
    class: 'starfighter',
    passengers: '0'
  },
  {
    name: 'Snowspeeder',
    model: 't-47 airspeeder',
    class: 'airspeeder',
    passengers: '0'
  },
  {
    name: 'TIE bomber',
    model: 'TIE/sa bomber',
    class: 'space/planetary bomber',
    passengers: '0'
  },
  {
    name: 'AT-AT',
    model: 'All Terrain Armored Transport',
    class: 'assault walker',
    passengers: '40'
  },
  {
    name: 'AT-ST',
    model: 'All Terrain Scout Transport',
    class: 'walker',
    passengers: '0'
  },
  {
    name: 'Storm IV Twin-Pod cloud car',
    model: 'Storm IV Twin-Pod',
    class: 'repulsorcraft',
    passengers: '0'
  },
  {
    name: 'Sail barge',
    model: 'Modified Luxury Sail Barge',
    class: 'sail barge',
    passengers: '500'
  }
];

export const fetchedPeopleData = {
  results: [
    {
      birth_year: '112BBY',
      created: '2014-12-10T15:10:51.357000Z',
      edited: '2014-12-20T21:17:50.309000Z',
      eye_color: 'yellow',
      films: (6)[
        ('https://swapi.co/api/films/2/',
        'https://swapi.co/api/films/5/',
        'https://swapi.co/api/films/4/',
        'https://swapi.co/api/films/6/',
        'https://swapi.co/api/films/3/',
        'https://swapi.co/api/films/1/')
      ],
      gender: 'n/a',
      hair_color: 'n/a',
      height: '167',
      homeworld: 'https://swapi.co/api/planets/1/',
      mass: '75',
      name: 'C-3PO',
      skin_color: 'gold',
      species: ['https://swapi.co/api/species/2/'],
      starships: [],
      url: 'https://swapi.co/api/people/2/',
      vehicles: []
    }
  ]
};

export const fetchedHomeWorldData = {
  climate: 'temperate',
  created: '2014-12-10T16:16:26.566000Z',
  diameter: '0',
  edited: '2014-12-20T20:58:18.452000Z',
  films: [],
  gravity: '1 standard',
  name: 'Tatooine',
  orbital_period: 'unknown',
  population: '200000',
  residents: ['https://swapi.co/api/people/10/'],
  rotation_period: 'unknown',
  surface_water: 'unknown',
  terrain: 'grass',
  url: 'https://swapi.co/api/planets/20/'
};

export const fetchedSpeciesData = {
  average_height: '180',
  average_lifespan: '120',
  classification: 'mammal',
  created: '2014-12-10T13:52:11.567000Z',
  designation: 'sentient',
  edited: '2015-04-17T06:59:55.850671Z',
  eye_colors: 'brown, blue, green, hazel, grey, amber',
  films: (7)[
    ('https://swapi.co/api/films/2/',
    'https://swapi.co/api/films/7/',
    'https://swapi.co/api/films/5/',
    'https://swapi.co/api/films/4/',
    'https://swapi.co/api/films/6/',
    'https://swapi.co/api/films/3/',
    'https://swapi.co/api/films/1/')
  ],
  hair_colors: 'blonde, brown, black, red',
  homeworld: 'https://swapi.co/api/planets/9/',
  language: 'Galactic Basic',
  name: 'Human',
  people: (35)[
    ('https://swapi.co/api/people/1/',
    'https://swapi.co/api/people/4/',
    'https://swapi.co/api/people/5/',
    'https://swapi.co/api/people/6/',
    'https://swapi.co/api/people/7/',
    'https://swapi.co/api/people/9/',
    'https://swapi.co/api/people/10/',
    'https://swapi.co/api/people/11/',
    'https://swapi.co/api/people/12/',
    'https://swapi.co/api/people/14/',
    'https://swapi.co/api/people/18/',
    'https://swapi.co/api/people/19/',
    'https://swapi.co/api/people/21/',
    'https://swapi.co/api/people/22/',
    'https://swapi.co/api/people/25/',
    'https://swapi.co/api/people/26/',
    'https://swapi.co/api/people/28/',
    'https://swapi.co/api/people/29/',
    'https://swapi.co/api/people/32/',
    'https://swapi.co/api/people/34/',
    'https://swapi.co/api/people/43/',
    'https://swapi.co/api/people/51/',
    'https://swapi.co/api/people/60/',
    'https://swapi.co/api/people/61/',
    'https://swapi.co/api/people/62/',
    'https://swapi.co/api/people/66/',
    'https://swapi.co/api/people/67/',
    'https://swapi.co/api/people/68/',
    'https://swapi.co/api/people/69/',
    'https://swapi.co/api/people/74/',
    'https://swapi.co/api/people/81/',
    'https://swapi.co/api/people/84/',
    'https://swapi.co/api/people/85/',
    'https://swapi.co/api/people/86/',
    'https://swapi.co/api/people/35/')
  ],
  skin_colors: 'caucasian, black, asian, hispanic',
  url: 'https://swapi.co/api/species/1/'
};
