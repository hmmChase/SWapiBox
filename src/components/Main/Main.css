main {
  padding: 20px;
  display: flex;
  justify-content: center;
  flex-direction: row;
  flex-wrap: wrap;
  

}

.film-crawl {
  text-align: center;
  width: 100%;
}

.star-wars {
  display: flex;
  overflow: hidden;
  /* justify-content: center; */
  /* position: relative; */
  height: 800px;
  color: #feda4a;
  font-family: 'Pathway Gothic One', sans-serif;
  font-size: 500%;
  font-weight: 600;
  letter-spacing: 6px;
  line-height: 150%;
  perspective: 300px;
  text-align: justify;
}

.crawl {
  position: relative;
  top: 99999px;
  transform-origin: 50% 100%;
  animation: crawl 60s linear;
}

article {
  height: 250px;
  width: 150px;
  color: white;
  /* background: lightblue; */
  margin: 10px;
  border: 2px solid yellow;
}

@keyframes crawl {
  0% {
    top: -100px;
    transform: rotateX(20deg) translateZ(0);
  }
  100% {
    top: -6000px;
    transform: rotateX(25deg) translateZ(-2500px);
  }
}

