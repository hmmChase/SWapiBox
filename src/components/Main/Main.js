import React, { Component } from 'react';
// import Card from '../Card/Card';
import './Main.css';

class Main extends Component {
  constructor(props) {
    super(props);

    this.state = {
    };
  }

  renderCategories = () => {
    if (this.props.category === "people") {
      return this.props.categoryData.map((person, index) => {
        return (
          <article key={index}>{person.name}</article>
        );

      });
    } else if (this.props.category === "planets") {
      return this.props.categoryData.map((planet, index) => {
        return (
          <article key={index}>{planet.name}</article>
        );
      });
    } else if (this.props.category === "vehicles") {
      return this.props.categoryData.map((vehicle, index) => {
        return (
          <article key={index}>{vehicle.name}</article>
        );
      });
    }
  }

  render() {
    return (
      <main>
        {
          !this.props.category ?
            <section className="film-crawl star-wars">
              <div className="crawl">
                <p>{this.props.randomFilmCrawl.crawl}</p>
                <p>{this.props.randomFilmCrawl.title}</p>
                <p>{this.props.randomFilmCrawl.date}</p>
              </div>
            </section>
            : null
        }
        {
          this.renderCategories()
        }
      </main>
    );
  }

}


export default Main;
