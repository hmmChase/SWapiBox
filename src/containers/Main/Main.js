import React from 'react';
import PropTypes from 'prop-types';
import Card from '../../components/Card/Card';
import Crawl from '../../components/Crawl/Crawl';
import './Main.css';

const Main = props => {
  const renderCategories = () => {
    return props.categoryData.map((cardObj, index) => {
      const favoriteCardNames = props.favorites.map(favorite => favorite.name);
      const favoriteBool = favoriteCardNames.includes(cardObj.name);

      return (
        <Card
          key={index}
          cardObj={cardObj}
          updateFavorites={props.updateFavorites}
          favoriteBool={favoriteBool}
        />
      );
    });
  };

  return (
    <main>
      {!props.category ? (
        <Crawl randomFilmCrawl={props.randomFilmCrawl} />
      ) : (
        renderCategories()
      )}
    </main>
  );
};



Main.propTypes = {
  category: PropTypes.string,
  categoryData: PropTypes.array,
  updateFavorites: PropTypes.func,
  randomFilmCrawl: PropTypes.object,
  favorites: PropTypes.array
};

export default Main;
