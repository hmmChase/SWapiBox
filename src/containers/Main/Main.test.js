import React from 'react';
import { shallow } from 'enzyme';
import Main from './Main';

import * as mockData from '../../__mocks__/mockData';

let mockProps = {
  category: '',
  categoryData: mockData.categoryData,
  favorites: [],
  randomFilmCrawl: mockData.randomFilmCrawl
};

describe('Main', () => {
  let main;

  beforeEach(() => {
    main = shallow(<Main {...mockProps} />);
  });

  it('matches the snapshot', () => {
    expect(main).toMatchSnapshot();
  });

  it('matches the snapshot with category selected', () => {
    mockProps.category = 'mock category';

    expect(main).toMatchSnapshot();
  });

  it('matches the snapshot with category selected and favorites added', () => {
    mockProps.category = 'mock category';
    mockProps.favorites = mockData.favorites;

    expect(main).toMatchSnapshot();
  });

  describe('renderCategories', () => {
    it.skip('returns a Card component for every cardObj', () => {
      expect(mockProps.categoryData).toHaveLength(10);

      // console.log(main.debug());
      // console.log(() => main.renderCategories());
      expect(() => main.renderCategories()).toHaveLength(10);
    });
  });
});
