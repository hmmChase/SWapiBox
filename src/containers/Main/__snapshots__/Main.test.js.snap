// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main
    category=""
    categoryData={
        Array [
            Object {
              "class": "wheeled",
              "model": "Digger Crawler",
              "name": "Sand Crawler",
              "passengers": "30",
            },
            Object {
              "class": "repulsorcraft",
              "model": "T-16 skyhopper",
              "name": "T-16 skyhopper",
              "passengers": "1",
            },
            Object {
              "class": "repulsorcraft",
              "model": "X-34 landspeeder",
              "name": "X-34 landspeeder",
              "passengers": "1",
            },
            Object {
              "class": "starfighter",
              "model": "Twin Ion Engine/Ln Starfighter",
              "name": "TIE/LN starfighter",
              "passengers": "0",
            },
            Object {
              "class": "airspeeder",
              "model": "t-47 airspeeder",
              "name": "Snowspeeder",
              "passengers": "0",
            },
            Object {
              "class": "space/planetary bomber",
              "model": "TIE/sa bomber",
              "name": "TIE bomber",
              "passengers": "0",
            },
            Object {
              "class": "assault walker",
              "model": "All Terrain Armored Transport",
              "name": "AT-AT",
              "passengers": "40",
            },
            Object {
              "class": "walker",
              "model": "All Terrain Scout Transport",
              "name": "AT-ST",
              "passengers": "0",
            },
            Object {
              "class": "repulsorcraft",
              "model": "Storm IV Twin-Pod",
              "name": "Storm IV Twin-Pod cloud car",
              "passengers": "0",
            },
            Object {
              "class": "sail barge",
              "model": "Modified Luxury Sail Barge",
              "name": "Sail barge",
              "passengers": "500",
            },
          ]
    }
    favorites={Array []}
    randomFilmCrawl={
        Object {
            "crawl": "Film Crawl",
            "date": "1982-09-18",
            "title": "Movie Title",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Crawl
        randomFilmCrawl={
                Object {
                        "crawl": "Film Crawl",
                        "date": "1982-09-18",
                        "title": "Movie Title",
                      }
        }
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "randomFilmCrawl": Object {
          "crawl": "Film Crawl",
          "date": "1982-09-18",
          "title": "Movie Title",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Crawl
          randomFilmCrawl={
                    Object {
                              "crawl": "Film Crawl",
                              "date": "1982-09-18",
                              "title": "Movie Title",
                            }
          }
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "randomFilmCrawl": Object {
            "crawl": "Film Crawl",
            "date": "1982-09-18",
            "title": "Movie Title",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Main matches the snapshot with category selected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main
    category=""
    categoryData={
        Array [
            Object {
              "class": "wheeled",
              "model": "Digger Crawler",
              "name": "Sand Crawler",
              "passengers": "30",
            },
            Object {
              "class": "repulsorcraft",
              "model": "T-16 skyhopper",
              "name": "T-16 skyhopper",
              "passengers": "1",
            },
            Object {
              "class": "repulsorcraft",
              "model": "X-34 landspeeder",
              "name": "X-34 landspeeder",
              "passengers": "1",
            },
            Object {
              "class": "starfighter",
              "model": "Twin Ion Engine/Ln Starfighter",
              "name": "TIE/LN starfighter",
              "passengers": "0",
            },
            Object {
              "class": "airspeeder",
              "model": "t-47 airspeeder",
              "name": "Snowspeeder",
              "passengers": "0",
            },
            Object {
              "class": "space/planetary bomber",
              "model": "TIE/sa bomber",
              "name": "TIE bomber",
              "passengers": "0",
            },
            Object {
              "class": "assault walker",
              "model": "All Terrain Armored Transport",
              "name": "AT-AT",
              "passengers": "40",
            },
            Object {
              "class": "walker",
              "model": "All Terrain Scout Transport",
              "name": "AT-ST",
              "passengers": "0",
            },
            Object {
              "class": "repulsorcraft",
              "model": "Storm IV Twin-Pod",
              "name": "Storm IV Twin-Pod cloud car",
              "passengers": "0",
            },
            Object {
              "class": "sail barge",
              "model": "Modified Luxury Sail Barge",
              "name": "Sail barge",
              "passengers": "500",
            },
          ]
    }
    favorites={Array []}
    randomFilmCrawl={
        Object {
            "crawl": "Film Crawl",
            "date": "1982-09-18",
            "title": "Movie Title",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Crawl
        randomFilmCrawl={
                Object {
                        "crawl": "Film Crawl",
                        "date": "1982-09-18",
                        "title": "Movie Title",
                      }
        }
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "randomFilmCrawl": Object {
          "crawl": "Film Crawl",
          "date": "1982-09-18",
          "title": "Movie Title",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Crawl
          randomFilmCrawl={
                    Object {
                              "crawl": "Film Crawl",
                              "date": "1982-09-18",
                              "title": "Movie Title",
                            }
          }
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "randomFilmCrawl": Object {
            "crawl": "Film Crawl",
            "date": "1982-09-18",
            "title": "Movie Title",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Main matches the snapshot with category selected and favorites added 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main
    category="mock category"
    categoryData={
        Array [
            Object {
              "class": "wheeled",
              "model": "Digger Crawler",
              "name": "Sand Crawler",
              "passengers": "30",
            },
            Object {
              "class": "repulsorcraft",
              "model": "T-16 skyhopper",
              "name": "T-16 skyhopper",
              "passengers": "1",
            },
            Object {
              "class": "repulsorcraft",
              "model": "X-34 landspeeder",
              "name": "X-34 landspeeder",
              "passengers": "1",
            },
            Object {
              "class": "starfighter",
              "model": "Twin Ion Engine/Ln Starfighter",
              "name": "TIE/LN starfighter",
              "passengers": "0",
            },
            Object {
              "class": "airspeeder",
              "model": "t-47 airspeeder",
              "name": "Snowspeeder",
              "passengers": "0",
            },
            Object {
              "class": "space/planetary bomber",
              "model": "TIE/sa bomber",
              "name": "TIE bomber",
              "passengers": "0",
            },
            Object {
              "class": "assault walker",
              "model": "All Terrain Armored Transport",
              "name": "AT-AT",
              "passengers": "40",
            },
            Object {
              "class": "walker",
              "model": "All Terrain Scout Transport",
              "name": "AT-ST",
              "passengers": "0",
            },
            Object {
              "class": "repulsorcraft",
              "model": "Storm IV Twin-Pod",
              "name": "Storm IV Twin-Pod cloud car",
              "passengers": "0",
            },
            Object {
              "class": "sail barge",
              "model": "Modified Luxury Sail Barge",
              "name": "Sail barge",
              "passengers": "500",
            },
          ]
    }
    favorites={Array []}
    randomFilmCrawl={
        Object {
            "crawl": "Film Crawl",
            "date": "1982-09-18",
            "title": "Movie Title",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Card
          cardObj={
                    Object {
                              "class": "wheeled",
                              "model": "Digger Crawler",
                              "name": "Sand Crawler",
                              "passengers": "30",
                            }
          }
          favoriteBool={false}
          updateFavorites={undefined}
/>,
        <Card
          cardObj={
                    Object {
                              "class": "repulsorcraft",
                              "model": "T-16 skyhopper",
                              "name": "T-16 skyhopper",
                              "passengers": "1",
                            }
          }
          favoriteBool={false}
          updateFavorites={undefined}
/>,
        <Card
          cardObj={
                    Object {
                              "class": "repulsorcraft",
                              "model": "X-34 landspeeder",
                              "name": "X-34 landspeeder",
                              "passengers": "1",
                            }
          }
          favoriteBool={false}
          updateFavorites={undefined}
/>,
        <Card
          cardObj={
                    Object {
                              "class": "starfighter",
                              "model": "Twin Ion Engine/Ln Starfighter",
                              "name": "TIE/LN starfighter",
                              "passengers": "0",
                            }
          }
          favoriteBool={false}
          updateFavorites={undefined}
/>,
        <Card
          cardObj={
                    Object {
                              "class": "airspeeder",
                              "model": "t-47 airspeeder",
                              "name": "Snowspeeder",
                              "passengers": "0",
                            }
          }
          favoriteBool={false}
          updateFavorites={undefined}
/>,
        <Card
          cardObj={
                    Object {
                              "class": "space/planetary bomber",
                              "model": "TIE/sa bomber",
                              "name": "TIE bomber",
                              "passengers": "0",
                            }
          }
          favoriteBool={false}
          updateFavorites={undefined}
/>,
        <Card
          cardObj={
                    Object {
                              "class": "assault walker",
                              "model": "All Terrain Armored Transport",
                              "name": "AT-AT",
                              "passengers": "40",
                            }
          }
          favoriteBool={false}
          updateFavorites={undefined}
/>,
        <Card
          cardObj={
                    Object {
                              "class": "walker",
                              "model": "All Terrain Scout Transport",
                              "name": "AT-ST",
                              "passengers": "0",
                            }
          }
          favoriteBool={false}
          updateFavorites={undefined}
/>,
        <Card
          cardObj={
                    Object {
                              "class": "repulsorcraft",
                              "model": "Storm IV Twin-Pod",
                              "name": "Storm IV Twin-Pod cloud car",
                              "passengers": "0",
                            }
          }
          favoriteBool={false}
          updateFavorites={undefined}
/>,
        <Card
          cardObj={
                    Object {
                              "class": "sail barge",
                              "model": "Modified Luxury Sail Barge",
                              "name": "Sail barge",
                              "passengers": "500",
                            }
          }
          favoriteBool={false}
          updateFavorites={undefined}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "cardObj": Object {
            "class": "wheeled",
            "model": "Digger Crawler",
            "name": "Sand Crawler",
            "passengers": "30",
          },
          "favoriteBool": false,
          "updateFavorites": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "cardObj": Object {
            "class": "repulsorcraft",
            "model": "T-16 skyhopper",
            "name": "T-16 skyhopper",
            "passengers": "1",
          },
          "favoriteBool": false,
          "updateFavorites": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "cardObj": Object {
            "class": "repulsorcraft",
            "model": "X-34 landspeeder",
            "name": "X-34 landspeeder",
            "passengers": "1",
          },
          "favoriteBool": false,
          "updateFavorites": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "function",
        "props": Object {
          "cardObj": Object {
            "class": "starfighter",
            "model": "Twin Ion Engine/Ln Starfighter",
            "name": "TIE/LN starfighter",
            "passengers": "0",
          },
          "favoriteBool": false,
          "updateFavorites": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "function",
        "props": Object {
          "cardObj": Object {
            "class": "airspeeder",
            "model": "t-47 airspeeder",
            "name": "Snowspeeder",
            "passengers": "0",
          },
          "favoriteBool": false,
          "updateFavorites": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "5",
        "nodeType": "function",
        "props": Object {
          "cardObj": Object {
            "class": "space/planetary bomber",
            "model": "TIE/sa bomber",
            "name": "TIE bomber",
            "passengers": "0",
          },
          "favoriteBool": false,
          "updateFavorites": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "6",
        "nodeType": "function",
        "props": Object {
          "cardObj": Object {
            "class": "assault walker",
            "model": "All Terrain Armored Transport",
            "name": "AT-AT",
            "passengers": "40",
          },
          "favoriteBool": false,
          "updateFavorites": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "7",
        "nodeType": "function",
        "props": Object {
          "cardObj": Object {
            "class": "walker",
            "model": "All Terrain Scout Transport",
            "name": "AT-ST",
            "passengers": "0",
          },
          "favoriteBool": false,
          "updateFavorites": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "8",
        "nodeType": "function",
        "props": Object {
          "cardObj": Object {
            "class": "repulsorcraft",
            "model": "Storm IV Twin-Pod",
            "name": "Storm IV Twin-Pod cloud car",
            "passengers": "0",
          },
          "favoriteBool": false,
          "updateFavorites": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "9",
        "nodeType": "function",
        "props": Object {
          "cardObj": Object {
            "class": "sail barge",
            "model": "Modified Luxury Sail Barge",
            "name": "Sail barge",
            "passengers": "500",
          },
          "favoriteBool": false,
          "updateFavorites": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Card
            cardObj={
                        Object {
                                    "class": "wheeled",
                                    "model": "Digger Crawler",
                                    "name": "Sand Crawler",
                                    "passengers": "30",
                                  }
            }
            favoriteBool={false}
            updateFavorites={undefined}
/>,
          <Card
            cardObj={
                        Object {
                                    "class": "repulsorcraft",
                                    "model": "T-16 skyhopper",
                                    "name": "T-16 skyhopper",
                                    "passengers": "1",
                                  }
            }
            favoriteBool={false}
            updateFavorites={undefined}
/>,
          <Card
            cardObj={
                        Object {
                                    "class": "repulsorcraft",
                                    "model": "X-34 landspeeder",
                                    "name": "X-34 landspeeder",
                                    "passengers": "1",
                                  }
            }
            favoriteBool={false}
            updateFavorites={undefined}
/>,
          <Card
            cardObj={
                        Object {
                                    "class": "starfighter",
                                    "model": "Twin Ion Engine/Ln Starfighter",
                                    "name": "TIE/LN starfighter",
                                    "passengers": "0",
                                  }
            }
            favoriteBool={false}
            updateFavorites={undefined}
/>,
          <Card
            cardObj={
                        Object {
                                    "class": "airspeeder",
                                    "model": "t-47 airspeeder",
                                    "name": "Snowspeeder",
                                    "passengers": "0",
                                  }
            }
            favoriteBool={false}
            updateFavorites={undefined}
/>,
          <Card
            cardObj={
                        Object {
                                    "class": "space/planetary bomber",
                                    "model": "TIE/sa bomber",
                                    "name": "TIE bomber",
                                    "passengers": "0",
                                  }
            }
            favoriteBool={false}
            updateFavorites={undefined}
/>,
          <Card
            cardObj={
                        Object {
                                    "class": "assault walker",
                                    "model": "All Terrain Armored Transport",
                                    "name": "AT-AT",
                                    "passengers": "40",
                                  }
            }
            favoriteBool={false}
            updateFavorites={undefined}
/>,
          <Card
            cardObj={
                        Object {
                                    "class": "walker",
                                    "model": "All Terrain Scout Transport",
                                    "name": "AT-ST",
                                    "passengers": "0",
                                  }
            }
            favoriteBool={false}
            updateFavorites={undefined}
/>,
          <Card
            cardObj={
                        Object {
                                    "class": "repulsorcraft",
                                    "model": "Storm IV Twin-Pod",
                                    "name": "Storm IV Twin-Pod cloud car",
                                    "passengers": "0",
                                  }
            }
            favoriteBool={false}
            updateFavorites={undefined}
/>,
          <Card
            cardObj={
                        Object {
                                    "class": "sail barge",
                                    "model": "Modified Luxury Sail Barge",
                                    "name": "Sail barge",
                                    "passengers": "500",
                                  }
            }
            favoriteBool={false}
            updateFavorites={undefined}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "cardObj": Object {
              "class": "wheeled",
              "model": "Digger Crawler",
              "name": "Sand Crawler",
              "passengers": "30",
            },
            "favoriteBool": false,
            "updateFavorites": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "cardObj": Object {
              "class": "repulsorcraft",
              "model": "T-16 skyhopper",
              "name": "T-16 skyhopper",
              "passengers": "1",
            },
            "favoriteBool": false,
            "updateFavorites": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "cardObj": Object {
              "class": "repulsorcraft",
              "model": "X-34 landspeeder",
              "name": "X-34 landspeeder",
              "passengers": "1",
            },
            "favoriteBool": false,
            "updateFavorites": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "function",
          "props": Object {
            "cardObj": Object {
              "class": "starfighter",
              "model": "Twin Ion Engine/Ln Starfighter",
              "name": "TIE/LN starfighter",
              "passengers": "0",
            },
            "favoriteBool": false,
            "updateFavorites": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "function",
          "props": Object {
            "cardObj": Object {
              "class": "airspeeder",
              "model": "t-47 airspeeder",
              "name": "Snowspeeder",
              "passengers": "0",
            },
            "favoriteBool": false,
            "updateFavorites": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5",
          "nodeType": "function",
          "props": Object {
            "cardObj": Object {
              "class": "space/planetary bomber",
              "model": "TIE/sa bomber",
              "name": "TIE bomber",
              "passengers": "0",
            },
            "favoriteBool": false,
            "updateFavorites": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "6",
          "nodeType": "function",
          "props": Object {
            "cardObj": Object {
              "class": "assault walker",
              "model": "All Terrain Armored Transport",
              "name": "AT-AT",
              "passengers": "40",
            },
            "favoriteBool": false,
            "updateFavorites": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "7",
          "nodeType": "function",
          "props": Object {
            "cardObj": Object {
              "class": "walker",
              "model": "All Terrain Scout Transport",
              "name": "AT-ST",
              "passengers": "0",
            },
            "favoriteBool": false,
            "updateFavorites": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "8",
          "nodeType": "function",
          "props": Object {
            "cardObj": Object {
              "class": "repulsorcraft",
              "model": "Storm IV Twin-Pod",
              "name": "Storm IV Twin-Pod cloud car",
              "passengers": "0",
            },
            "favoriteBool": false,
            "updateFavorites": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "9",
          "nodeType": "function",
          "props": Object {
            "cardObj": Object {
              "class": "sail barge",
              "model": "Modified Luxury Sail Barge",
              "name": "Sail barge",
              "passengers": "500",
            },
            "favoriteBool": false,
            "updateFavorites": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
